abel/linbasex.py

```diff
@@ -0,0 +1,418 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import numpy as np
+import abel
+import scipy as sci
+from scipy.special import eval_legendre
+from scipy import ndimage
+
+###############################################################################
+# linbasex - inversion procedure based on 1-dimensional projections of
+#            velocity-map images
+#
+# As described in:
+#   Gerber, Thomas, Yuzhu Liu, Gregor Knopp, Patrick Hemberger, Andras Bodi,
+#   Peter Radi, and Yaroslav Sych,
+#     “Charged Particle Velocity Map Image Reconstruction with One-Dimensional
+#      Projections of Spherical Functions.”
+#     Review of Scientific Instruments 84, no. 3 (March 1, 2013):
+#                                      033101–033101 – 10.
+#     doi:10.1063/1.4793404.
+#
+# 2016-04- Thomas Gerber and Daniel Hickstein - theory and code updates
+# 2016-04- Stephen Gibson core code extracted from the supplied jupyter
+#          notebook (see #167: https://github.com/PyAbel/PyAbel/issues/167)
+#
+###############################################################################
+
+_linbasex_parameter_docstring = \
+    r"""Inverse Abel transform using 1d projections of VM-images.
+
+    `Gerber, Thomas, Yuzhu Liu, Gregor Knopp, Patrick Hemberger, Andras Bodi,
+    Peter Radi, and Yaroslav Sych.
+    Charged Particle Velocity Map Image Reconstruction with One-Dimensional Projections of Spherical Functions.` Review of Scientific Instruments 84, no. 3 (March 1, 2013): 033101–033101 – 10.
+    <http://dx.doi.org/10.1063/1.4793404>`_
+
+     linbasex determines the contribution of spherical functions,
+     calculating the weight of each :math:`Y_{i0}`. The reconstructed 3D object
+     is obtained by adding all the :math:`Y_{i0}` contributions, from which
+     slices are derived.
+
+
+    Parameters
+    ----------
+    Dat: numpy 2D array
+        image data must be square shape of odd size
+    an: list
```

Comment:
In general, I would like to give these variables more informative names. `an -> angles` or `projection_angles`. `un -> order` or `polynomial order`. `inc -> radial_step`. `sig_s -> smoothing`.